app/controllers/concerns/accounts/authorization.rb

```diff
@@ -0,0 +1,185 @@
+# OpenProject is an open source project management software.
+# Copyright (C) 2010-2024 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+# ++
+
+##
+# Intended to be used by the ApplicationController to provide authorization helpers
+module Accounts::Authorization
+  extend ActiveSupport::Concern
+
+  METHODS_ENFORCING_AUTHORIZATION = %i[require_admin authorize authorize_global authorize_in_optional_project].freeze
+
+  included do
+    class_attribute :authorization_ensured,
+                    default: {
+                      only: [],
+                      except: [],
+                      generally_allowed: false,
+                      controller: self
+                    }
+  end
+
+  private
+
+  def authorization_check_required
+    unless authorization_is_ensured?(params[:action])
+      if Rails.env.development?
+        raise <<-MESSAGE
+          Authorization check required for #{params[:action]} in #{self.class.name}.
+
+          Use any method of 'authorize', 'authorize_global', 'authorize_in_optional_project'
+          or 'require_admin' to ensure authorization. If authorization is checked by any other means,
+          affirm the same by calling 'authorization_checked!' in the controller. If the authorization does
+          not need to be checked for this action, affirm the same by calling 'no_authorization_required!'
+        MESSAGE
+      else
+        render_403
+      end
+    end
+  end
+
+  # Authorize the user for the requested controller action.
+  # To be used in before_action hooks
+  def authorize(ctrl = params[:controller], action = params[:action])
+    do_authorize({ controller: ctrl, action: }, global: false)
+  end
+
+  # Authorize the user for the requested controller action outside a project
+  # To be used in before_action hooks
+  def authorize_global
+    action = { controller: params[:controller], action: params[:action] }
```

Comment:
Done in [d60076c](https://github.com/opf/openproject/pull/15733/commits/d60076c725491f8652e967f9ba23d0320db8d443)

Since both methods are only used as before actions I removed the parameters for both